swap/int256/int256.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package int256
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Int256 represents an signed integer of 256 bits
+type Int256 struct {
+	value big.Int
+}
+
+var minInt256 = new(big.Int).Mul(big.NewInt(-1), new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil)) // -(2^255)
+var maxInt256 = new(big.Int).Sub(new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil), big.NewInt(1))  // 2^255 - 1
+
+// NewInt256 creates a Int256 struct with an initial underlying value of the given param
+// returns an error when the result falls outside of the signed 256-bit integer range
+func NewInt256(value big.Int) (*Int256, error) {
+	u := new(Int256)
+	return u.set(value)
+}
+
+// Int256From creates a Int256 struct based on the given int64 param
+// any int64 is valid as a Int256
+func Int256From(base int64) *Int256 {
+	u := new(Int256)
+	u.value = *new(big.Int).SetInt64(base)
+	return u
+}
+
+// Copy creates and returns a new Int256 instance, with its underlying value set matching the receiver
+func (u *Int256) Copy() *Int256 {
+	v := new(Int256)
+	v.value = *new(big.Int).Set(&u.value)
+	return v
+}
+
+// Value returns the underlying private value for a Int256 struct
+func (u *Int256) Value() big.Int {
+	return u.value
```

Comment:
if you change `u.value` to be a pointer, and get rid of all of those excess allocations in all of the helper functions of this type, you could just dereference `u.value` when returning the value in this function, causing the desired effect while mitigating all this unnecessary overhead.

Reply:
addressed